src/prefect/engine/cloud/flow_runner.py

```diff
@@ -154,7 +251,9 @@ def initialize_run(  # type: ignore
         """
 
         # load id from context
-        flow_run_id = prefect.context.get("flow_run_id")
+        flow_run_id = prefect.context.get("flow_run_id") or context.get(
```

Comment:
I recommend reversing these - the passed context should take precedence over the global context

Reply:
Removed the check the passed context entirely. I'm not sure how I originally found the case where the `flow_run_id` wasn't in the global context, but I can't reproduce it.